README.adoc

```diff
@@ -1,88 +1,323 @@
+All you need to build these projects is Java 8.0 (Java SDK 1.8) or later and Maven 3.3.1 or later.
 
-// Suggested Approach to the Quickstarts
-include::shared-doc/suggested-approach.adoc[leveloffset=+1]
+[[prerequisites]]
+== Prerequisites
 
-[[system_requirements]]
-== System Requirements
+Before you begin, you must perform the following tasks.
 
-The applications these projects produce are designed to be run on {productNameFull} {productVersion} or later.
+. xref:build_wildfly_server_and_boms[Build the WildFly server and BOMs]: Fetch the latest WildFly source code and build the WildFly Server and the WildFly BOMs. This installs the Maven artifacts that are required by the WildFly quickstarts.
+
+. xref:install_the_quickstart_parent_artifact_in_maven[Install the Quickstart Parent Artifact in Maven]: The root `POM.xml` file defines dependencies that are required by some of the quickstarts.
+
+. xref:build_quickstart_readme_files[Build the Quickstart README.html Files]: The quickstart `README` files are written in AsciiDoc to provide modular, reusable content; however, this makes them difficult to read. For this reason, you also must build the quickstart `README.html` files from the AsciiDoc source.
+
+[[build_wildfly_server_and_boms]]
+=== Build the WildFly Server and BOMs
+
+You must build the WildFly server and BOMs to install the Maven artifacts before you can build a quickstart or a quickstart's `README.html` file. The WildFly development team normally takes care of building and publishing the Maven artifacts needed to build the quickstarts in this repository, but as with all new development, there might be times where you need a library that has not yet been updated. Occasionally you might also need an update that was made to WildFly core.
+
+The order for building the WildFly server is as follows.
+
+. xref:build_wildfly_core[Build WildFly core] (if required)
+. xref:build_wildfly_server[Build the Wildfly server]
+. xref:build_wildfly_boms[Build the Wildfly BOMs]
+
+[discrete]
+[[build_wildfly_core]]
+==== Build WildFly Core
+
+IMPORTANT: This is only necessary if you need a recent update that was made to WildFly core but it is not yet published.
+
+. If you have not yet done so, you must clone the https://github.com/wildfly/wildfly-core[Wildfly core] repository and navigate to it. You might also want to change the remote name from `origin` to `upstream` to be consistent with your other repositories.
++
+[source,options="nowrap"]
+----
+$ git clone git@github.com:wildfly/wildfly-core.git
+$ cd wildfly-core
+$ git remote rename origin upstream
+----
+. Verify that your local `master` branch contains the latest updates.*
++
+[source,options="nowrap"]
+----
+$ git fetch upstream
+$ git checkout master
+$ git reset --hard upstream/master
+----
+. Build WildFly core using the following command.
++
+[source,options="nowrap"]
+----
+$ mvn clean install
+----
+
+[discrete]
+[[build_wildfly_server]]
+==== Build the WildFly Server
```

Comment:
When reading I get the impression that I always need to build BOMs, when that is true only when building a Quickstarts development branch (e.g. master) AND it depends on a SNAPSHOT BOMs version ( e.g. https://github.com/wildfly/quickstart/blob/master/pom.xml#L69 ). Maybe rework this content as troubleshooting when build fails?

Reply:
Agree. I have reworked this.